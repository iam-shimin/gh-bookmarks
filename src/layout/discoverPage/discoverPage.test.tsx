import userEvent from '@testing-library/user-event';
import React from 'react';
import renderWithStoreAndRouter from 'utils/render';
import DiscoverPage from './discoverPage';
// import repoService from 'services/repos';
jest.mock('services/repos');

describe('<DiscoverPage />', () => {
	test('match snapshot', () => {
		const { asFragment } = renderWithStoreAndRouter(<DiscoverPage />);
		expect(asFragment()).toMatchSnapshot();
	});

	test('searchable', async () => {
		Object.defineProperty(Object, 'fromEntries', {
			get: () => (entries: [string, any]) => {
				const res: {[_: string]: any} = {};
				for (let [key, value] of entries) {
					res[key] = value;
				}
				return res;
			}
		});
		const { getByRole, findAllByRole } = renderWithStoreAndRouter(<DiscoverPage />);
		const txtBox = getByRole('searchbox');
		userEvent.type(txtBox, 'a');
		const results = await findAllByRole('heading');
		expect(results.length).toEqual(2);
	})
})