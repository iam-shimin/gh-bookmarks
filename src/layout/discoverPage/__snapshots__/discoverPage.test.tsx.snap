// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscoverPage /> match snapshot 1`] = `
<DocumentFragment>
  <header>
    <div
      class="titlebar"
    >
      <a
        class="titlebar-backbtn util--no-underline"
        href="/"
        title="Go Back to Bookmarks"
      >
        &lt;
      </a>
      <span
        class="titlebar-title"
      >
        Discover
      </span>
    </div>
    <form
      class="searchbar"
      role="search"
    >
      <input
        aria-label="Search GitHub for Repositories"
        name="q"
        placeholder="Search GitHub for Repositories"
        type="search"
      />
      <select
        name="type"
        title="Search by Type"
      >
        <option
          value="user"
        >
          Username
        </option>
        <option
          selected=""
          value="repo"
        >
          Repository
        </option>
      </select>
      <button
        class="btn-plain btn-search"
      >
        <span
          aria-label="Search"
          role="img"
        >
          üîç
        </span>
      </button>
    </form>
  </header>
  <section
    class="container-wrapper"
  >
    <div
      class="fade alert alert-primary show"
      role="alert"
    >
      Start by searching.
    </div>
  </section>
  <footer>
    <span>
      Github Bookmarks
    </span>
    <div>
      <a
        href="https://github.com/iam-shimin/gh-bookmarks"
        rel="noopener noreferrer"
        target="_blank"
      >
        Fork this repo
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
